applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing_symmetric.h

```diff
@@ -0,0 +1,188 @@
+// ==============================================================================
+//  KratosShapeOptimizationApplication
+//
+//  License:         BSD License
+//                   license: ShapeOptimizationApplication/license.txt
+//
+//  Main authors:    Armin Geiser, https://github.com/armingeiser
+//
+// ==============================================================================
+
+#ifndef MAPPER_VERTEX_MORPHING_SYMMETRIC_H
+#define MAPPER_VERTEX_MORPHING_SYMMETRIC_H
+
+// System includes
+#include <iostream>
+#include <string>
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "spatial_containers/spatial_containers.h"
+#include "spaces/ublas_space.h"
+#include "mapper_base.h"
+#include "filter_function.h"
+#include "symmetry_base.h"
+
+// ==============================================================================
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+*/
+
+class KRATOS_API(SHAPE_OPTIMIZATION_APPLICATION) MapperVertexMorphingSymmetric : public Mapper
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    // Type definitions for better reading later
+    typedef Node < 3 > NodeType;
+    typedef Node < 3 > ::Pointer NodeTypePointer;
+    typedef std::vector<NodeType::Pointer> NodeVector;
+    typedef std::vector<NodeType::Pointer>::iterator NodeIterator;
+    typedef std::vector<double>::iterator DoubleVectorIterator;
+    typedef ModelPart::ConditionsContainerType ConditionsArrayType;
+    typedef array_1d<double,3> array_3d;
+
```

Comment:
Not important but for consistency you can add  `Type` to all the typedef values like `NodeVector`  --> `NodeVectorType`